src/Statistics.jl

```diff
@@ -348,172 +337,313 @@ function median!(
-    var(M, x, m=nothing; corrected=true, kwargs...)
-
-compute the variance of a `Vector` `x` of `n` data points on the
-[`Manifold`](@ref) `M`, i.e.
-
-````math
-\frac{1}{c} \sum_{i=1}^n d_{\mathcal M}^2 (x_i,m),
-````
-where `c` is a correction term, see
-[Statistics.var](https://juliastats.org/StatsBase.jl/stable/scalarstats/#Statistics.var).
-and `m` is the provideed mean of `x`. The uncorrected variance
-can be activated by setting `corrected=false`.
-"""
 function var(
     M::Manifold,
     x::AbstractVector,
-    m = nothing;
+    m;
     corrected::Bool = true,
-    kwargs...
 )
-    if (m === nothing)
-        m = mean(M, x; kwargs...)
-    end
     n = length(x)
     w = _unit_weights(n)
-    return var(M, x, w, m; corrected = corrected, kwargs...)
+    return var(M, x, w, m; corrected = corrected)
 end
 
+var(M::Manifold, x::AbstractVector, w::AbstractWeights; kwargs...) = mean_and_var(M, x, w; kwargs...)[2]
+var(M::Manifold, x::AbstractVector; kwargs...) = mean_and_var(M, x; kwargs...)[2]
+
 @doc doc"""
-    std(M, x, w::AbstractWeights, m=nothing; corrected=false, kwargs...)
+    std(M, x, m=mean(M, x); corrected=true, kwargs...)
+    std(M, x, w::AbstractWeights, m=mean(M, x, w); corrected=false, kwargs...)
 
-compute the (weighted) standard deviation of a `Vector` `x` of `n` data points on the
-[`Manifold`](@ref) `M`, i.e.
+compute the optionally weighted standard deviation of a `Vector` `x` of `n` data
+points on the [`Manifold`](@ref) `M`, i.e.
 
 ````math
 \sqrt{\frac{1}{c} \sum_{i=1}^n w_i d_{\mathcal M}^2 (x_i,m)},
 ````
 where `c` is a correction term, see
-[Statistics.var](https://juliastats.org/StatsBase.jl/stable/scalarstats/#Statistics.var).
-The (weighted) mean of `x` can be specified as `m`, and the corrected variance
-can be activated by setting `corrected=true`. All further `kwargs...` are passed
-to the computation of the mean (if that is not provided).
+[Statistics.std](https://juliastats.org/StatsBase.jl/stable/scalarstats/#Statistics.std).
+The mean of `x` can be specified as `m`, and the corrected variance
+can be activated by setting `corrected=true`.
 """
-function std(
-    M::Manifold,
-    x::AbstractVector,
-    w::AbstractWeights,
-    m = nothing;
-    corrected::Bool = false,
-    kwargs...
-)
-    return sqrt(var(M, x, w, m; corrected = corrected, kwargs...))
-end
+std(M::Manifold, args...; kwargs...) = sqrt(var(M, args...; kwargs...))
 
 @doc doc"""
-    std(M, x, m=nothing; corrected=true, kwargs...)
+    cov(M, x, m=mean(M, x); corrected=true, kwargs...)
+    cov(M, x, w::AbstractWeights, m=mean(M, x, w); corrected=false, kwargs...)
 
-compute the variance of a `Vector` `x` of `n` data points on the
-[`Manifold`](@ref) `M`, i.e.
+Compute the (optionally weighted) covariance of a `Vector` `x` of `n` data
+points on the [`Manifold`](@ref) `M`, i.e.
 
 ````math
-\sqrt{\frac{1}{c} \sum_{i=1}^n d_{\mathcal M}^2 (x_i,m)},
+\frac{1}{c} \sum_{i=1}^n w_i (\log_{\mu} x_i) (\log_{\mu} x_i)^\mathrm{T},
```

Comment:
This might be my only problem with all these nice commits here: What's the log-transposed? I think in general you can only do that if you have a basis (ONB) and the log returns a vector representation with respect to that basis (compare `Grassmann` for example where the tangent space does not have such a unique nice representer. What's otherwise the transpose?

Reply:
This concept of covariance (as a bilinear form independent of ONB) is borrowed from [Pennec's paper](http://www-sop.inria.fr/asclepios/Publications/Xavier.Pennec/Pennec.JMIV06.pdf). You bring up some good points, and the more I think about it the less it makes sense. I'll need to think about this some more.

Reply:
I think in the embedding (for example with the Figure 2 of the Sphere) that makes totally sense, I just haven't read the paper close enough to make sure he is just taking about that. If he means the general case – you can of course use it, Xavier Pennec knows his stuff :)

Reply:
After reading a bit more, this seems complicated, and I'd like to work it out in another PR after we have good support for ONBs. I'll remove `cov` from this PR.

From what I can tell, the expression for covariance assumes one is in an exponential chart not necessarily given an ONB. But it seems the expression A) assumes the tangent vector representation is a column vector and B) assumes a Euclidean metric (it notes that the trace of the covariance matrix is the Riemannian variance, which is only always true when the inner product is Euclidean, which in an exponential chart happens when using an ONB).

This interpretation is bolstered by [this paper](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.475.1066&rep=rep1&type=pdf), where Pennec works out an example of the Riemannian normal on covariance matrices and defines the same covariance with respect to an orthonormal basis.